node_modules/gh-orgunit/tests/test-import.js

```diff
@@ -774,6 +774,71 @@ describe('Organisational units - import', function() {
                 });
             });
         });
+
+        /**
+         * Test that verifies that the correct groups are created
+         */
+        it('verify group creation', function(callback) {
+            // Create a course
+            OrgUnitTestsUtil.assertCreateOrgUnit(global.tests.admins.cam2013.client, 'Test course', 'course', {}, function(course) {
+
+                // Import 2 parts, each with 2 modules, 2 series and 2 event
+                var importData = _getTestTree('1course-2subject-2part-2module-2series-2event');
+                OrgUnitTestsUtil.assertImportOrgUnit(global.tests.admins.cam2013.client, course.id, importData, false, function() {
+                    OrgUnitTestsUtil.assertExportOrgUnit(global.tests.admins.cam2013.client, course.id, null, function(dataPostFirstImport) {
+
+                        OrgUnitTestsUtil.getAllChildren(global.tests.admins.cam2013.client, course.AppId, [course.id], function(orgUnits, series) {
+                            // There should be 14 child organisation units in total. Two subjects
+                            // holding 2 parts with 2 modules each
+                            var subjects = _.filter(orgUnits, {'type': 'subject'});
+                            var parts = _.filter(orgUnits, {'type': 'part'});
+                            var modules = _.filter(orgUnits, {'type': 'module'});
+                            assert.strictEqual(subjects.length, 2);
+                            assert.strictEqual(parts.length, 4);
+                            assert.strictEqual(modules.length, 8);
+                            assert.strictEqual(orgUnits.length, 14);
+
+                            var subjectGroupIds = _.chain(subjects).pluck('GroupId').uniq().value().sort();
+                            var partGroupIds = _.chain(parts).pluck('GroupId').uniq().value().sort();
+                            var moduleGroupIds = _.chain(modules).pluck('GroupId').uniq().value().sort();
+
+                            // Each subject should have a different group id
+                            assert.strictEqual(subjectGroupIds.length, 2);
+
+                            // Each part should have a different group id
+                            assert.strictEqual(partGroupIds.length, 4);
+
+                            // Parts should NOT inherit the subject or course's group id
+                            assert.strictEqual(_.intersection([course], subjectGroupIds).length, 0);
+                            assert.strictEqual(_.intersection([course], partGroupIds).length, 0);
```

Comment:
It looks like we are intersecting an array with one course *object* with a list of string ids, that can't be right?